clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -890,8 +890,8 @@ public void run() {
                         if (coordinatorUnknown()) {
                             if (findCoordinatorFuture == null)
                                 lookupCoordinator();
-                            else
-                                AbstractCoordinator.this.wait(retryBackoffMs);
+                            // coordinator may still be unknown, instead of an immediate retry, let's wait for a bit.
+                            AbstractCoordinator.this.wait(retryBackoffMs);
```

Comment:
Instead of waiting in all cases, I'm wondering if it makes sense to check the result of `lookupCoordinator`. If the future has failed, then we backoff.